dali/base/dasds.cpp

```diff
@@ -1757,14 +1752,15 @@ void buildNotifyData(MemoryBuffer &notifyData, PDState state, CPTStack *stack, M
                 PTree &child = stack->item(s);
                 const char *str = child.queryName();
                 notifyData.append(strlen(str), str);
-                if (child.queryParent())
+                unsigned pos = child.querySiblingPosition();
+                if (0 == pos) // common case
+                    notifyData.append(3, "[1]");
+                else
                 {
                     char temp[12];
-                    unsigned written = numtostr(temp, parent->findChild(&child)+1);
+                    unsigned written = numtostr(temp, pos+1);
                     notifyData.append('[').append(written, temp).append(']');
```

Comment:
trivial: Wondering why this doesn't use append(pos+1).  And that function could probably be improved.

Reply:
I have no idea.. I'll change it.

Reply:
now I remember!
It's because the append(unsigned) , is to a MemoryBuffer, so the change is now appending the binary value, not the string as intended..